authz/rbac_translator.go

```diff
@@ -266,6 +281,48 @@ func parseRules(rules []rule, prefixName string) (map[string]*v3rbacpb.Policy, e
 	return policies, nil
 }
 
+func parseAuditLoggingOptions(options auditLoggingOptions) (*v3rbacpb.RBAC_AuditLoggingOptions, error) {
+	optionsRbac := v3rbacpb.RBAC_AuditLoggingOptions{}
+	if options.AuditCondition == "" && len(options.AuditLoggers) == 0 {
+		// No audit logger parsed - fine
+		return &optionsRbac, nil
+	} else if options.AuditCondition == "" {
+		return &optionsRbac, fmt.Errorf("AuditLogger config present but missing AuditCondition")
+	} else if len(options.AuditLoggers) == 0 {
+		return &optionsRbac, fmt.Errorf("AuditCondition present but missing AuditLogger config")
+	}
+
+	switch options.AuditCondition {
+	case "NONE":
```

Comment:
The generated Go pkg from the proto should have a map to help you convert a string to the enum (which is int32 in Go) directly. You can consider using that, which feels more robust.

Reply:
Ah, I was thinking that _should_ exist, I just didn't immediately see it. Good catch! Fixed